source/minar.cpp

```diff
@@ -251,7 +251,7 @@ int minar::SchedulerData::start(){
 
             if(dispatch_tree.get_num_elements() > 0) {
                 CallbackNode *root = dispatch_tree.get_root();
-                now_plus_tolerance = wrapTime(now + root->tolerance);
+                now_plus_tolerance = wrapTime(now + root->tolerance/2);
```

Comment:
That's why I asked about the intention of double sided tolerance. When we store it we times it by 2 and every time we use it we divide. This is not intuitive unless there is a higher design purpose for this. This fix is the minimum amount of code change, but I can equally change all the other place to use single sided tolerance.